hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -234,13 +238,26 @@ public BackgroundTaskResult call() throws Exception {
               if (!(toReclaim.getKeyInfosCount() ==
                   repeatedOmKeyInfo.getOmKeyInfoList().size())) {
                 toReclaimList.add(toReclaim.build());
+                toNextDBList.add(toNextDb.build());
+              } else {
+                // The key can be reclaimed here.
+                List<BlockGroup> blocksForKeyDelete = omSnapshot
+                    .getMetadataManager()
+                    .getBlocksForKeyDelete(deletedKey);
+                if (blocksForKeyDelete != null) {
+                  keysToPurge.addAll(blocksForKeyDelete);
+                }
               }
-              toNextDBList.add(toNextDb.build());
               deletionCount++;
             }
             // Submit Move request to OM.
             submitSnapshotMoveDeletedKeys(snapInfo, toReclaimList,
                 toNextDBList);
+
+            // Delete keys From deletedTable
+            long startTime = Time.monotonicNow();
+            processKeyDeletes(keysToPurge, omSnapshot.getKeyManager(),
+                startTime, snapInfo.getTableKey());
```

Comment:
Feels like we can put the `startTime` inside `processKeyDeletes()` and remove it from the params list.